templates/default/systemd.erb

```diff
@@ -4,10 +4,11 @@ After=network.target
 Wants=network.target
 
 [Service]
+EnvironmentFile=-/etc/sysconfig/<%= @name %>
```

Comment:
For sure this is possible.
But handling an environment file is in that case even better because we don't need to reload systemctl if there is any change in the env variables.